api/clients/v2/disperser_client.go

```diff
@@ -1,4 +1,4 @@
-package clients
+package v2
```

Comment:
How about keeping the package name `clients` here?
Its typical for golang modules to use an "invisible" version suffix (this is actually part of how golang's algo does version management). See for eg this urfave example: https://github.com/urfave/cli/blob/main/examples/example-cli/example-cli.go (imported as cli/v3, but module is just used as `cli.`, not `v3.` which would be awkward).

I'm not sure if its typical to do the same thing for packages, but I feel like it would be cleaner?

Reply:
To do this, don't you need to version the entire module (like [here](https://github.com/urfave/cli/blob/main/go.mod#L1) in your example)?

Reply:
No I'm just saying leave the directory structure the same (`api/clients/v2/.*.go`), but name the package `clients` (package can be named different than the directory).